File: classe_note/frontend/src/components/forms/ClasseService.js
// src/components/forms/ClasseService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/api/classes';

export default {
  getAllClasses() {
    return axios.get(API_URL);
  },

  getClasseById(id) {
    return axios.get(`${API_URL}/${id}`);
  },

  createClasse(classe) {
    return axios.post(API_URL, classe);
  },

  updateClasse(id, classe) {
    return axios.put(`${API_URL}/${id}`, classe);
  },

  deleteClasse(id) {
    return axios.delete(`${API_URL}/${id}`);
  }
};

File: classe_note/backend/src/main/java/com/iba/cahierdetexte/service/ClasseService.java
package com.iba.cahierdetexte.service;

import com.iba.cahierdetexte.model.Classe;
import java.util.List;

public interface ClasseService {
    Classe createClasse(Classe classe);
    Classe getClasseById(Long id);
    List<Classe> getAllClasses();
    Classe updateClasse(Long id, Classe classe);
    void deleteClasse(Long id);
}
